src/server.rs

```diff
@@ -330,6 +341,19 @@ impl HttpServer {
         )
     }
 
+    /// poll event use mio poll method, and handle Interrupted error explictly
+    fn poll_events(&mut self, events: &mut mio::Events) -> Result<()> {
+        loop {
+            if let Err(e) = self.poll.poll(events, None) {
+                if e.kind() == ErrorKind::Interrupted {
```

Comment:
Need to judge WouldBlock here?

Reply:
yes, probably yes, according to document from mio, we should handle WouldBlock error, and should not treat this as an error
> If operation fails with [WouldBlock](https://doc.rust-lang.org/nightly/std/io/error/enum.ErrorKind.html#variant.WouldBlock), then the caller should not treat this as an error, but instead should wait until another readiness event is received.